src/pybind/mgr/dashboard/frontend/src/app/shared/models/cd-notification.ts

```diff
@@ -0,0 +1,40 @@
+import { NotificationType } from '../enum/notification-type.enum';
+
+export class CdNotification {
+  message: string;
+  timestamp: string;
+  title: string;
+  type: NotificationType;
+
+  constructor(type?: NotificationType, message?: string, title?: string) {
+    this.type = type;
+    this.message = message;
+    this.title = title;
+
+    /* string representation of the Date object so it can be directly compared
+    with the timestamps parsed from localStorage */
+    this.timestamp = new Date().toJSON();
+  }
+
+  textClass() {
+    switch (this.type) {
```

Comment:
What happens if ``type`` is not set (it's an optional parameter regarding the constructor)? Wouldn't it make sense to add a default case?